### PR TITLE
documentation: add CONTRIBUTING.md file #60

### PR DESCRIPTION
**Pull request by vraybaud**

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+# Contributing
+
+Stellio is an open source project part of FIWARE. We welcome any contributions from everyone.
```
Comment (bobeal) [inline docs/CONTRIBUTING.md:3]:
```suggestion
Stellio is an open source project, currently in the incubator to be become a FIWARE Generic Enabler. We welcome any contributions from everyone.  
```

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
+# Contributing
+
+Stellio is an open source project part of FIWARE. We welcome any contributions from everyone.  
+If you wish to contribute on our code, you should first read and follow guidelines mentionned in this document.
```
Comment (bobeal) [inline docs/CONTRIBUTING.md:4]:
```suggestion
If you wish to contribute on our code, you should first read and follow guidelines mentioned in this document.
```

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+## Ask a question 
+
+Please if you want to ask a question regarding Stellio and how to use it. Create an issue
```
Comment (bobeal) [inline docs/CONTRIBUTING.md:8]:
```suggestion
Please if you want to ask a question regarding Stellio and how to use it, create an issue
```

File: docs/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+- Version of the project linked to the issue.
+- Explain expected and actual behaviour.
+- Steps to reproduce the issue.
+
+There are many labels available, please feel free to put a label on your issue.
+
+## Requesting new features
+
+Stellio is implementing the NGSI-LD API specification.  You may find unsupported features.
```
Comment (bobeal) [inline docs/CONTRIBUTING.md:24]:
```suggestion
Stellio implements the NGSI-LD API specification.  You may find unsupported features.  
```

**Comment by vraybaud** [comment]
I have read the CLA Document and I hereby sign the CLA

File: docs/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+## Project Structure
+
+Stellio is divided into five main modules: api-gateway, entity-service, search-service, subscription-service and shared.
+Each module follows the same schema :
+- **src**:
+    - **main**: contains the source code.
+        - **kotlin**: contains the source kotlin code.
+        - **resources** : contains the resources.
+    - **test**: contains the test code, a unit test class of a source code class should have the same package.
```
Comment (HoucemKacem) [inline docs/CONTRIBUTING.md:83]: Maybe you can mention that the test folder contains also two sub-folders Kotlin and resources.
And the resources contains the test data.

File: docs/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
+# Contributing
+
+Stellio is an open source project, currently in the incubator to become a FIWARE Generic Enabler. We welcome any contributions from everyone.  
+If you wish to contribute on our code, you should first read and follow guidelines mentioned in this document.
+
+## Ask a question 
+
+Please if you want to ask a question regarding Stellio and how to use it, create an issue
+ in [Stackoverflow](https://stackoverflow.com/) with the tag fiware-stellio.
```
Comment (HoucemKacem) [inline docs/CONTRIBUTING.md:9]: I remember seeing this Tag when preparing the Fiwrare requirements, do we need to prepare something for it ?

File: docs/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+## Making Changes
+
+Contributions to Stellio are done using a pull request. Find below a description on how to do a good pull request :
+
+- Fork this repository
+- Create a branch on the forked repository.  
+A good branch name should follow this schema : `<label>/<issue_number>-<name>`
+
+    - `<label>` is the label of your issue : feature, bug, refactor, documentation...
```
Comment (HoucemKacem) [inline docs/CONTRIBUTING.md:39]: fix or bug ?

File: docs/CONTRIBUTING.md (hunk 6 above)
Comment (vraybaud) [reply docs/CONTRIBUTING.md:9]: I don't know, I saw that in the requirements too. I guess we have to subscribe to this tag to see questions related to Stellio.

File: docs/CONTRIBUTING.md (hunk 7 above)
Comment (vraybaud) [reply docs/CONTRIBUTING.md:39]: fix, indeed

File: docs/CONTRIBUTING.md (hunk 5 above)
Comment (vraybaud) [reply docs/CONTRIBUTING.md:83]: Yep Indeed, the api-gateway module is not containing any resources folder but this is not a matter I think I can add the sub-folders here..

**Comment by vraybaud** [comment]
I have read the CLA Document and I hereby sign the CLA

File: docs/CONTRIBUTING.md (hunk 6 above)
Comment (bobeal) [reply docs/CONTRIBUTING.md:9]: If you find a way to subscribe to a tag on SO, I'd like to know how :)

Comment (vraybaud) [reply docs/CONTRIBUTING.md:9]: With RSS you can subscribe to 
```https://stackoverflow.com/feeds/tag/fiware-stellio```
But first we need to create a question to generate the tag.

Comment (vraybaud) [reply docs/CONTRIBUTING.md:9]: I found also in Stackoverflow that it should be working with "Watched Tags", but I ask to see it working. (just subscribed so I don't know if it's going to work) from https://meta.stackoverflow.com/questions/254318/how-to-subscribe-to-tags
image

**Pull request merged**